src/openms/source/TRANSFORMATIONS/FEATUREFINDER/ElutionModelFitter.cpp

```diff
@@ -96,20 +96,18 @@ double ElutionModelFitter::calculateFitQuality_(const TraceFitter* fitter,
   double rt_end = min(fitter->getUpperRTBound(),
                       traces[0].peaks.back().first);
 
-  for (MassTraces::const_iterator tr_it = traces.begin();
-       tr_it != traces.end(); ++tr_it)
+  for (const auto& trace : traces)
   {
-    for (vector<pair<double, const Peak1D*> >::const_iterator p_it =
-           tr_it->peaks.begin(); p_it != tr_it->peaks.end(); ++p_it)
+    for (const auto& peak : trace.peaks)
```

Comment:
here I would definitely use a structure binding for better readability:
consider:
for (const auto &[rt, const Peak*] : trace.peaks)

this will yield shorter and more readable code